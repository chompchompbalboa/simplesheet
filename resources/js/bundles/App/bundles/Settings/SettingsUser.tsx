//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'

import ContentContent from '@app/bundles/Content/ContentContent'
import SettingsGroup from '@app/bundles/Settings/SettingsGroup'
import SettingsUserEmail from '@app/bundles/Settings/SettingsUserEmail'
import SettingsUserName from '@app/bundles/Settings/SettingsUserName'
import SettingsUserSubscription from '@app/bundles/Settings/SettingsUserSubscription'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SettingsUser = () => {

  return (
    <ContentContent>
      <SettingsGroup
        header="Profile">
        <SettingsUserName />
        <SettingsUserEmail />
      </SettingsGroup>
      <SettingsGroup
        header="Billing">
        <SettingsUserSubscription />
      </SettingsGroup>
    </ContentContent>
  )
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SettingsUser
