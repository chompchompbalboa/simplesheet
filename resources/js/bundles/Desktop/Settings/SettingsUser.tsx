//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'

import ContentContent from '@desktop/Content/ContentContent'
import SettingsGroup from '@desktop/Settings/SettingsGroup'
import SettingsUserEmail from '@desktop/Settings/SettingsUserEmail'
import SettingsUserName from '@desktop/Settings/SettingsUserName'
import SettingsUserSubscription from '@desktop/Settings/SettingsUserSubscription'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SettingsUser = () => {

  return (
    <ContentContent>
      <SettingsGroup
        header="Profile">
        <SettingsUserName />
        <SettingsUserEmail />
      </SettingsGroup>
      <SettingsGroup
        header="Subscription">
        <SettingsUserSubscription />
      </SettingsGroup>
    </ContentContent>
  )
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SettingsUser
