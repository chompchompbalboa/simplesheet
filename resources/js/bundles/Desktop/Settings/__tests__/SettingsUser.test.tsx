//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import axiosMock from 'axios'

import { renderWithRedux } from '@/testing/library'
import { appStateFactory, IAppStateFactoryInput } from '@/testing/mocks/appState'

import SettingsUser from '@desktop/Settings/SettingsUser'

//-----------------------------------------------------------------------------
// Setup
//-----------------------------------------------------------------------------
const {
  allFiles,
  allFileIds,
  allSheetsFromDatabase
} = appStateFactory({} as IAppStateFactoryInput)

const fileId = allFileIds[0]
const sheetId = allFiles[fileId].typeId
const sheetFromDatabase = allSheetsFromDatabase[sheetId]

// @ts-ignore
axiosMock.get.mockResolvedValue({ data: sheetFromDatabase })

console.warn = jest.fn()
console.error = jest.fn()

//-----------------------------------------------------------------------------
// Tests
//-----------------------------------------------------------------------------
describe('SettingsUser', () => {
  
  it("renders without crashing", async () => {
    const container = renderWithRedux(<SettingsUser />)
    expect(container).toMatchSnapshot()
  })
  
  it("displays the 'Profile' section", async () => {
    const { queryByText } = renderWithRedux(<SettingsUser />)
    expect(queryByText('Profile')).toBeTruthy()
  })
  
  it("displays the 'Subscription' section", async () => {
    const { queryByText } = renderWithRedux(<SettingsUser />)
    expect(queryByText('Subscription')).toBeTruthy()
  })

})
