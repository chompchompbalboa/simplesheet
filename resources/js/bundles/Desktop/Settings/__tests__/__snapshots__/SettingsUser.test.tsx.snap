// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsUser renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  height: 100%;
  padding: 0.75rem;
}

.c1 {
  width: 100%;
  background-color: rgb(245,245,245);
  border-radius: 3px;
  margin-bottom: 0.75rem;
}

.c2 {
  cursor: pointer;
  padding: 0.5rem 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(225,225,225);
  color: rgb(10,10,10);
  font-size: 0.95rem;
  font-weight: bold;
  border: 1px solid rgb(175,175,175);
  border-bottom: none;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c3 {
  padding: 0.5rem;
  display: block;
  border: 1px solid rgb(175,175,175);
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
}

.c4 {
  margin-bottom: 0.5rem;
  padding: 0.25rem 0.125rem;
  width: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgb(50,50,50);
  border-bottom: 1px dashed rgb(200,200,200);
}

.c5 {
  display: inline;
  font-weight: bold;
  font-size: 0.85rem;
  white-space: nowrap;
}

.c6 {
  width: 100%;
  text-align: right;
  border: none;
  outline: none;
  background-color: transparent;
  font-family: inherit;
  font-size: 0.8rem;
  color: rgb(50,50,50);
}

.c9 {
  cursor: pointer;
  min-width: 6rem;
  padding: 0.5rem;
  white-space: nowrap;
  border: 1px solid rgb(150,150,150);
  outline: 0;
  background-color: userColorPrimary;
  color: white;
  font-size: 0.75rem;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s;
  transition: all 0.15s;
  opacity: 1;
}

.c9:hover {
  background-color: rgb(0,150,25);
  border: 1px solid rgb(0,150,25);
}

.c7 {
  width: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  color: rgb(200,0,0);
  padding-right: 0.5rem;
}

.c10 {
  padding: 0.125rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
              
            Profile
          </div>
          <div
            class="c3"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
              >
                Name:
              </label>
              <input
                class="c6"
                type="text"
                value=""
              />
            </div>
            <div
              class="c4"
            >
              <label
                class="c5"
              >
                Email:
              </label>
              <input
                class="c6"
                disabled=""
                type="text"
                value=""
              />
            </div>
            <form>
              <div
                class="c4"
              >
                <label
                  class="c5"
                >
                  Enter Current Password:
                </label>
                <input
                  class="c6"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="c4"
              >
                <label
                  class="c5"
                >
                  Enter New Password:
                </label>
                <input
                  class="c6"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="c4"
              >
                <label
                  class="c5"
                >
                  Confirm New Password:
                </label>
                <input
                  class="c6"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="c7"
              >
                <div
                  class="c8"
                />
                <div
                  class=""
                >
                  <button
                    class="c9"
                  >
                    Change Password
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
              
            Subscription
          </div>
          <div
            class="c3"
          >
            <div
              class="c10"
            >
              <div
                class=""
              >
                Your lifetime subscription was succesfully purchased on 
                <b>
                  February 19th, 2020
                </b>
                . Thank you for choosing Tasksheet!
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC gfuBZH"
    >
      <div
        class="sc-AykKD fiaMDY"
      >
        <div
          class="sc-AykKE eRmdkC"
        >
            
          Profile
        </div>
        <div
          class="sc-AykKF ezKywo"
        >
          <div
            class="sc-AykKG hTatXH"
          >
            <label
              class="sc-AykKH byzbga"
            >
              Name:
            </label>
            <input
              class="sc-AykKI cJyGpj"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-AykKG hTatXH"
          >
            <label
              class="sc-AykKH byzbga"
            >
              Email:
            </label>
            <input
              class="sc-AykKI cJyGpj"
              disabled=""
              type="text"
              value=""
            />
          </div>
          <form>
            <div
              class="sc-AykKG hTatXH"
            >
              <label
                class="sc-AykKH byzbga"
              >
                Enter Current Password:
              </label>
              <input
                class="sc-AykKI cJyGpj"
                type="password"
                value=""
              />
            </div>
            <div
              class="sc-AykKG hTatXH"
            >
              <label
                class="sc-AykKH byzbga"
              >
                Enter New Password:
              </label>
              <input
                class="sc-AykKI cJyGpj"
                type="password"
                value=""
              />
            </div>
            <div
              class="sc-AykKG hTatXH"
            >
              <label
                class="sc-AykKH byzbga"
              >
                Confirm New Password:
              </label>
              <input
                class="sc-AykKI cJyGpj"
                type="password"
                value=""
              />
            </div>
            <div
              class="sc-AykKK gEJuCH"
            >
              <div
                class="sc-fzXfLO jGDPaO"
              />
              <div
                class="sc-fzXfLP efXezz"
              >
                <button
                  class="sc-AykKJ dVgYjO"
                >
                  Change Password
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="sc-AykKD fiaMDY"
      >
        <div
          class="sc-AykKE eRmdkC"
        >
            
          Subscription
        </div>
        <div
          class="sc-AykKF ezKywo"
        >
          <div
            class="sc-fzXfNN kOUuGD"
          >
            <div
              class="sc-fzXfNM eleAIA"
            >
              Your lifetime subscription was succesfully purchased on 
              <b>
                February 19th, 2020
              </b>
              . Thank you for choosing Tasksheet!
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
